### PR TITLE
Add support for fetching license info from github

### PR DESCRIPTION
**Pull request by nicarl**
### Summary of changes
Adds support to fetch license information based on a github URL from the github API.

### How can the changes be tested
- Execute unit tests
- open the app and fill e.g. `https://github.com/opossum-tool/OpossumUI` in the URL
Screenshot_20220207_191609

### CODE REVIEW

File: src/Frontend/Components/FetchLicenseInformationButton/license-fetching-helpers.ts [hunk 1]
```diff
@@ -6,9 +6,14 @@
...
 import { convertPypiPayload, getPypiAPIUrl } from './pypi-fetching-helpers';
 import { convertNpmPayload, getNpmAPIUrl } from './npm-fetching-helpers';
+import {
+  convertGithubPayload,
+  getGithubAPIUrl,
+} from './github-fetching-helpers';
 
 const PYPI_REGEX = new RegExp('^https://pypi.org/(pypi|project)/[\\w-+,_]+/?$');
 const NPM_REGEX = new RegExp('^https://npmjs.com/(package/)?[\\w-+,_@/]+/?$');
+const GITHUB_REGEX = new RegExp('^https://github.com/');
```
Comment (nicarl) [inline src/Frontend/Components/FetchLicenseInformationButton/license-fetching-helpers.ts:16]: The regex is not perfect yet, it should have at least two elements separated by `/` after `github.com`. I will still add this

File: src/Frontend/Components/FetchLicenseInformationButton/github-fetching-helpers.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  return {
+    namespace: urlSuffix[0],
+    packageName: urlSuffix[1],
+  };
+}
+
+const GITHUB_SCHEMA: Schema = {
+  type: 'object',
+  properties: {
+    html_url: { type: 'string' },
```
Comment (benedikt-richter) [inline src/Frontend/Components/FetchLicenseInformationButton/github-fetching-helpers.ts:34]: As we later cast to string, can this be required?

File: src/Frontend/Components/FetchLicenseInformationButton/github-fetching-helpers.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    },
+  },
+  required: ['license', 'html_url'],
+};
+
+export async function convertGithubPayload(
+  payload: Response
+): Promise<PackageInfo> {
+  const convertedPayload = await payload.json();
+  jsonSchemaValidator.validate(convertedPayload, GITHUB_SCHEMA, {
```
Comment (benedikt-richter) [inline src/Frontend/Components/FetchLicenseInformationButton/github-fetching-helpers.ts:49]: After validation, can we could cast the ´convertedPayload´ to the respective type in order not to have to cast individual fields later.

File: src/Frontend/Components/FetchLicenseInformationButton/github-fetching-helpers.ts [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    throwError: true,
+  });
+
+  const { namespace, packageName } =
+    getPackageNamespaceAndPackageNameFromGithubURL(
+      convertedPayload.html_url as string
+    );
+  const licenseText = convertedPayload.content
+    ? Buffer.from(convertedPayload.content, 'base64').toString()
+    : null;
```
Comment (benedikt-richter) [inline src/Frontend/Components/FetchLicenseInformationButton/github-fetching-helpers.ts:59]: Could we take `undefined` here and get rid of `??` in line 62?

**Comment by benedikt-richter** [comment]
Works!

I noted that the license text is retrieved as well and is displayed as "modified license text". Not sure this can be avoided, as probably we cannot be sure in general that it's an unmodified text...

**Review by leslielazzarino — APPROVED** [review]
Niiice

**Comment by nicarl** [comment]
> Works!
> 
> I noted that the license text is retrieved as well and is displayed as "modified license text". Not sure this can be avoided, as probably we cannot be sure in general that it's an unmodified text...

I think there are two alternatives:
1. Check if the license text received from github is matching the text from frequent licenses
2. Ignore the license text from github

Do you have any opinions?

**Review by leslielazzarino — APPROVED** [review]

**Review by benedikt-richter — APPROVED** [review]

**Pull request merged**